tests/integration/specs/common/autostore.spec.js

```diff
@@ -0,0 +1,79 @@
+import { expect } from 'chai';
+import nock from 'nock';
+import { init, DataStore, DataStoreType, User, Query } from '__SDK__';
+import { collectionName } from '../config';
+import { cleanUpAppData } from '../utils';
+
+describe('AutoStore', function() {
+  const cachedDocs = [];
+
+  before(function () {
+    // Turn off nock
+    return nock.restore();
+  });
+
+  before(function() {
+    // Initialize the SDK
+    return init({
+      appKey: process.env.APP_KEY,
+      appSecret: process.env.APP_SECRET
+    });
+  });
+
+  before(function () {
+    // Remove any existing data in the collection
+    return cleanUpAppData(collectionName);
+  });
+
+  before(function() {
+    // Signup an anonymous user
+    return User.signup();
+  });
+
+  before(function() {
+    const collection = DataStore.collection(collectionName, DataStoreType.Sync);
+    return collection.save({ title: 'Cached Book' })
+      .then((doc) => {
+        cachedDocs.push(doc);
+        const query = new Query().equalTo('_id', doc._id);
+        collection.clearSync(query);
+      });
+  });
+
+  before(function() {
+    // Turn on nock
+    return nock.activate();
+  });
+
+  after(function () {
+    // Turn off nock
+    return nock.restore();
+  });
+
+  after(function() {
+    // Remove any existing data in the collection
+    return cleanUpAppData(collectionName);
+  });
+
+  after(function() {
+    // Remove the active user
+    const activeUser = User.getActiveUser();
+    if (activeUser) {
+      return User.remove(activeUser._id, { hard: true });
+    }
+    return null;
+  });
+
+  it('should return cached items when the network request fails', async function() {
+    const scope = nock('https://baas.kinvey.com')
+      .get(`/appdata/${process.env.APP_KEY}/${collectionName}`)
+      .replyWithError({
+        message: 'no network connection'
+      });
+
+    const collection = DataStore.collection(collectionName, DataStoreType.Auto);
+    const docs = await collection.find();
+    expect(docs).to.deep.equal(cachedDocs);
+    expect(scope.isDone()).to.equal(true);
+  });
```

Comment:
I think we should add at least a few more tests for this iteration of the feature - an error is returned when there is an actual problem non-connection related (no permission for example) and one for the happy path for **save**, **create, update, remove, findById, removeById** operations. For these operations, also a test where connection error appears and we make sure that the operations are reflected in the offline provider and the kinvey_sync. Many of the tests can easily be added if the appropriate suites are extended a bit. For example **common-crud.spec.js** has a **dataStoreTypes** array, which can be extended with the AutoStore. Of course behavior for AS would need to be tweaked in some tests, but this should save a lot of work and time.